### PR TITLE
Simplify contract generation.

### PR DESCRIPTION
**Pull request by samth**

### CODE REVIEW

File: typed-racket-lib/typed-racket/private/type-contract.rkt [hunk 1]
```diff
@@ -449,8 +449,7 @@
                  (not (regexp? v)))
             (flat/sc #`(quote #,v))
             (flat/sc #`(flat-named-contract '#,v (lambda (x) (equal? x '#,v))) v))]
-       [(Base-name/contract: sym ctc)
-        (flat/sc #`(flat-named-contract '#,sym (flat-contract-predicate #,ctc)) sym)]
+       [(Base-name/contract: sym ctc) (flat/sc ctc)]
```
Comment (bennn) [inline typed-racket-lib/typed-racket/private/type-contract.rkt]: Why not `(flat/sc ctc sym)` ?

File: typed-racket-lib/typed-racket/private/type-contract.rkt [hunk 2]
```diff
@@ -1013,7 +1012,8 @@
       (prefix-in t: (types numeric-predicates))))
   (provide (all-defined-out))
 
-  (define-syntax-rule (numeric/sc name body)
+  (define-syntax-rule (numeric/sc name body) (flat/sc #'body)
```
Comment (bennn) [inline typed-racket-lib/typed-racket/private/type-contract.rkt]: Why not `(flat/sc #'body 'name)` ?

**Comment by bennn** [comment]
The "else" case for `Struct:` around line 700 can be simpler too, I think. From:

```
(flat/sc #`(flat-named-contract '#,(syntax-e pred?) (lambda (x) (#,pred? x))))
```

to

```
(flat/sc pred?)
```

**Comment by samth** [comment]
A few motivations:

- @rfindler persuaded me a while back that we should show the actual contracts.
- This avoids a bunch of allocation.
- This allows inlining a bunch of contracts, reducing the number of pointless definitions.
- Combining these makes the generated contracts in the expanded code easier to read (the immediate motivation for me).

**Comment by bennn** [comment]
ping @samth : were you planning to add more to this PR?

**Comment by samth** [comment]
I think this is ready now.

**Pull request merged**